app/grandchallenge/algorithms/views.py

```diff
@@ -111,12 +112,49 @@ class AlgorithmList(FilterMixin, PermissionListMixin, ListView):
     }
     ordering = "-created"
     filter_class = AlgorithmFilter
+    paginate_by = 40
+
+    @property
+    def _current_page(self):
+        return int(self.request.GET.get("page", 1))
+
+    @property
+    def _filters_applied(self):
+        return any(k for k in self.request.GET if k.lower() != "page")
+
+    def _get_page(self):
+        int_qs = super().get_queryset().order_by("-created")
+        self.int_filter = AlgorithmFilter(self.request.GET, int_qs,)
+
+        total_count = int_qs.count()
+
+        int_paginator = Paginator(self.int_filter.qs, self.paginate_by)
+
+        num_pages = int_paginator.num_pages
+        num_results = int_paginator.count
+
+        try:
+            int_page = int_paginator.page(self._current_page)
+        except EmptyPage:
+            int_page = []
+
+        return int_page, num_pages, num_results, total_count
```

Comment:
This seems to be more or less the same as the code added to the reader study list. Could this be consolidated into either an existing or a new mixin?

Reply:
Yes, true that. I just added a new mixin. Will change the reader study code once this and the other PR are merged.

Reply:
And if _that's_ true, maybe go one step further and change the html to a partial that you include in each page that uses it.